Makefile

```diff
@@ -13,6 +13,7 @@ lint:
 	uv run --no-sync ruff check src tests examples || exit 1
 	uv run --no-sync --all-extras mypy src tests examples || exit 1
 	uv run --no-sync --all-extras pyright src tests examples || exit 1
+	uv run --no-sync --all-extras basedpyright examples -p examples/pyrightconfig.json || exit 1
```

Comment:
The reason for using basedpyright instead of pyright: pyright has a bug that manifests in this check - basedpyright has fixed this bug
```
test_kitchen_sink/example.py:43:5 - error: Type of "flags" is partially unknown
    Type of "flags" is "list[Unknown]" (reportUnknownVariableType)
```

If you want to switch the other check to basedpyright, so you don't deal with pyright anymore, we would need to provide a configuration, since basedpyright has different defaults.
Let me know if you would prefer that.